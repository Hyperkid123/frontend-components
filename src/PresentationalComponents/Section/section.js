import React from 'react';
import propTypes from 'prop-types';
import classNames from 'classnames';

import './section.scss';

/**
 * This is a dumb component that only recieves properties from a smart component.
 * Dumb components are usually functions and not classes.
 *
 * Section Modifiers:
 *  Types = 'content'
 */

const Section = ({ type, children, className, ...props }) => {

    let sectionClasses = classNames(
        className,
        { [`ins-l-${type}`]: type !== undefined }
    );

     return (
        <section {...props} className={sectionClasses}> {children} </section>
    );
};

export default Section;

Section.propTypes = {
  type: propTypes.string,
  className: propTypes.string,
  children: propTypes.any.isRequired
};
