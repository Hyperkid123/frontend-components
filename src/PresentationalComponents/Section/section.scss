@import './src/Utilities.scss';

section.ins-l-content {
  @include rem('padding', $ins-padding);
}

section.ins-l-button-group {
  > * { display: inline; }
  * + * { @include rem('margin-left', 5px); }
  @include rem('margin', $ins-margin 0);
}

section.ins-l-icon-group {
  * + * { margin-left: 7.5px; }
}

section.ins-l-icon-group__with-major {
  * + * { margin-left: 7.5px; }
  .ins-battery:last-of-type {
    padding-left: 15px;
    border-left: 2px solid #eaeaea;
    span.label{
      font-weight: 500;
      margin: 0 10px;
    }
  }
}