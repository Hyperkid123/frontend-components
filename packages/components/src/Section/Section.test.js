import React from 'react';
import Section from './Section';
import { render } from '@testing-library/react';

describe('Section component', () => {
  it('should render correctly', () => {
    const { container } = render(<Section type="page">Test</Section>);
    expect(container).toMatchSnapshot();
  });
});
