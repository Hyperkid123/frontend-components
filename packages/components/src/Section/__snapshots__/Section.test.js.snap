// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Section component should render correctly 1`] = `
<div>
  <section
    class="ins-l-page"
  >
     
    Test
     
  </section>
</div>
`;
